packages/permissions/src/hooks/use-is-authorized/use-is-authorized.js

```diff
@@ -0,0 +1,28 @@
+import React from 'react';
+import { Context } from '@commercetools-frontend/application-shell-connectors';
+import {
+  hasSomePermissions,
+  hasEveryPermissions,
+} from '../../utils/has-permissions';
+
+// Forward-compatibility with React Hooks 🎉
+const useIsAuthorized = React.useContext
+  ? ({ demandedPermissions, shouldMatchSomePermissions = false }) => {
+      const applicationContext = React.useContext(Context);
+      const actualPermissions = applicationContext.permissions;
+
+      const isAuthorized = shouldMatchSomePermissions
+        ? hasSomePermissions(demandedPermissions, actualPermissions)
+        : hasEveryPermissions(demandedPermissions, actualPermissions);
+
+      return isAuthorized;
+    }
+  : () => {
+      throw new Error(
+        `React hooks do not seem to be available yet in the installed React version "${
```

Comment:
🆒

Reply:
We do that on other hooks too. I didn't care to test it again. In another hook there is a teset.